homeassistant/util/dt.py

```diff
@@ -322,16 +322,16 @@ def find_next_time_expression_time(
             now += dt.timedelta(seconds=1)
             continue
 
-        now_is_ambiguous = _datetime_ambiguous(now)
-        result_is_ambiguous = _datetime_ambiguous(result)
+        if not _datetime_ambiguous(now):
+            return result
```

Comment:
All the ifs below checked `now_is_ambiguous` so we can return right away if its not